extensions/interactions/ImageClickInput/ImageClickInput.js

```diff
@@ -92,6 +92,12 @@ oppia.directive('oppiaInteractiveImageClickInput', [
               rulesService: imageClickInputRulesService
             });
           };
+          $scope.$on('playerStateChange', function(evt, newStateName) {
```

Comment:
This is ingenious! I think it's too indirect, though. It's weird that only one interaction depends on playerTranscriptService and the others don't.

Also, I noticed a potential bug. I created an exploration with two states, with the first involving an image click interaction. I solved the first question and moved to the second state, then used the navigation dots at the top to move back to the first state. The highlights still show up despite the interaction being inactive.

I think we'll need to solve the problem of "inactive interactions" in a more general way. Instead of using this, could we pass in an arg `isActive` through the directive (so it's a sibling of onSubmit) and have the interaction use $scope.isActive() to determine whether it's active or not? Then the interaction would change its display based on the value of $scope.isActive().

I also think it probably makes sense to do just the collection text change in this PR, and leave the interaction stuff for another PR (possibly after the upcoming release, since I suspect it might take a little bit of time to get right). Would that be OK?

Thanks!